include/picongpu/particles/atomicPhysics2/initElectrons/Inelastic2BodyCollisionFromCoMoving.hpp

```diff
@@ -0,0 +1,278 @@
+
+            //      choose direction
+            /// @todo avoid calculating theta in the 2d case
+            float_X const u = rngGenerator();
+            float_X const v = rngGenerator();
+
+            float_X const cosTheta = 1._X - 2._X * v;
+            float_X const sinTheta = math::sqrt(v * (2._X - v));
+            float_X const phi = 2._X * static_cast<float_X>(picongpu::PI) * u;
+
+            float_X sinPhi;
+            float_X cosPhi;
+            pmacc::math::sincos(phi, sinPhi, cosPhi);
+
+            float3_64 const directionVector = float3_64(sinTheta * cosPhi, sinTheta * sinPhi, cosTheta);
+
+            // UNIT_MASS * UNIT_LENGTH / UNIT_TIME, weighted
+            auto momentumStarElectron_IonSystem = MatrixVector(normMomentumStarElectron_IonSystem * directionVector);
+
+            //      Lorentz transformation from IonSystem to LabSystem
+            // square of original momentum of the ion in Lab-System
+            // UNIT_MASS^2 * UNIT_LENGTH^2 / UNIT_TIME^2, not weighted
+            float_64 const momentumSquaredIon_LabSystem
+                = static_cast<float_64>(pmacc::math::l2norm2(ion[momentum_]) / (ion[weighting_] * ion[weighting_]));
+
+            // unitless
+            float3_64 betaIonSystem;
+            // unitless
+            float_64 normSquaredBetaIon_LabSystem;
+
+            if(momentumSquaredIon_LabSystem == 0._X)
+            {
+                /// @note momentum is always 3 dimensional in picongpu
+                betaIonSystem = float3_64(0., 0., 0.);
+                normSquaredBetaIon_LabSystem = 0.;
+            }
+            else
+            {
+                // square of original beta of the ion in Lab-System
+                // unitless
+                // beta^2 = 1/(1 + (m^2*c^2)/p^2)
+                //  unitless + (UNIT_MASS^2 * UNIT_SPEED^2)(not weighted)
+                //  /( UNIT_MASS^2 * UNIT_SPEED^2)(not weighted) = unitless
+                normSquaredBetaIon_LabSystem = 1. / (1. + mSquaredCSquaredIon / momentumSquaredIon_LabSystem);
+
+                betaIonSystem =
+                    // magnitude
+                    math::sqrt(normSquaredBetaIon_LabSystem)
+                    // direction
+                    * static_cast<float3_64>(ion[momentum_] / pmacc::math::l2norm(ion[momentum_]));
+            }
+
+            // unitless
+            auto beta = MatrixVector(betaIonSystem);
+
+            float_64 const gammaIonSystem = math::sqrt(momentumSquaredIon_LabSystem / mSquaredCSquaredIon + 1.);
+
+            // lower 3x3 block of Lorentz transformation matrix for transformation from
+            // Ion-System to Lab-System
+            Matrix_3x3 lorentzMatrix;
+            fillLorentzMatrix(beta, gammaIonSystem, normSquaredBetaIon_LabSystem, lorentzMatrix);
+
+            //      space components of Lorentz boost
+            MatrixVector momentumStarElectron_LabSystem;
```

Comment:
This instance is not initialized because internal MAtrix is using an pmacc::Array which is not initializing its data

Reply:
yes, which is why I am initializing it in the line below